### ISSUE TITLE
Red pane after saving in editors

### ISSUE THREAD

**Issue by SabineMa** [guillep/pharo-core#15106]
**Bug description**
After saving in any editor in Pharo 11, a part or the whole pane of the editor is red and this window can no longer be used. There was a discussion in Discord:
https://discord.com/channels/223421264751099906/278558427796602882/1164880827310669865
https://discord.com/channels/223421264751099906/278558427796602882/1166305029926420550

**To Reproduce**
Unfortunately not clear at the moment.
Possibly in conjunction with cmd-s for saving.

**Expected behavior**
No red window:-)

**Screenshots**
![IMG_0212](https://github.com/pharo-project/pharo/assets/13763544/a1522cb7-755e-4df7-a02a-e7b70c6b4c7b)

**Version information:**
 - OS: Mac but also win -see discord
 - Pharo Version 11

**Comment by SabineMa** [comment]
This is from Yanni:
yanni — gestern um 16:59 Uhr
I see the red box too. I'm on MacOS. It happened often prior to Pharo10. I was on Pharo10 for a long time (one year plus), and rarely saw the red box. After moving to Pharo11 a few months ago, I see it now and then (compared to almost never with Pharo10). It usually happens after a code change - I usually use cmd-S keyboard shortcut to save, not the menu. I try to avoid making code changes in the Debugger, because I've found that the change sometimes gets "lost" - i.e. proceeding from debugger fixes the problem, but when I browse the method from System browser, the code is nowhere to be found (not even in method version history).

My guess is that there is some timing issue with more than one process updating something, when a code change is saved - which leads to loss of code and/or redraw problems. My image had become sluggish recently, and I couldn't find any extra Process instances running, nor was the image extra big. But I was not getting red boxes when doing a code save. Instead, I got debugger windows with stacktrace in some window redraw code. I just closed the extra debugger. I still have that sluggish image, but the debugger would pop up now and then - no idea how to make it happen.

Norbert Hartl and Carlos reported that they have it, too.

**Comment by Rinzwind** [comment]
This seems closely related to issue #9638 for which a fix was committed in pull request #14853.

**Comment by MarcusDenker** [comment]
I backported two fixes, I don't think they are fixing this problem, but for sure nice to have:

https://github.com/pharo-project/pharo/pull/15374
https://github.com/pharo-project/pharo/pull/15373

**Comment by astares** [comment]
I only had this issue in the debugger editing pane when clicking too fast, was easy to patch - but this is solved with backport of #15372.

Any chance to put a *one time halt* into one of the drawing methods and try to debug the case?

**Comment by tinchodias** [comment]
I agree with with @Rinzwind that this could be fixed by enclosing some code into a `defer: [ ... ]` as in #14853. 
The cause might be that a "non-ui process" mutates the Calypso morph in an unexpected point of the "ui process".

**Comment by SabineMa** [comment]
we are moving our staging environment from P8 to P11.
the staging environment is hosted in a docker container on a Synology NAS. 
I can report that this issue with the red containers occurs much more than in other settings.

![Bildschirmfoto 2024-11-11 um 11 19 57](https://github.com/user-attachments/assets/0ca4510a-65c5-4db4-9cbc-60b3f5a94498)

I don't know it it is related but the processor on the container is than higher than normal and. 
The image is often frozen then.

**Comment by astares** [comment]
@SabineMa 
I can just guess but to me it seems like a bug in UI initialization or drawing logic where (one or more) submorphs might be "nil" and not respond to #extent and no proper extent is available to proceed.

The problem then often is that the image freezes in UI mode/redrawing or iteration on submorphs when it happens.
It might be called over and over again when the next debugger opens - so CPU goes up and UI is not responsive and tools like debugger might be hard to use if you end up in such situation.

What you could try to do is to manipulate the image before the loading problem appeary and implement
#extent in UndefinedObject to at least return same valid extent and see if UI at minimum becomes responsible again to be able to debug the issue.

**Comment by SabineMa** [comment]
thank you @astares for the first run the issue seems to be fixed for our staging with the workaround..
